src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -239,6 +252,85 @@ public boolean hasParserResults() {
+            for (BibEntry entry : dataBase.getEntries()) {
+                writeXMPtoPDFsOfEntry(databaseContext, entry.getCitationKey().orElse("<no cite key defined>"), entry, encoding, filePreferences, xmpPdfExporter);
+            }
+            return;
+        }
+
+        Vector<String> citeKeys = new Vector<>();
+        Vector<String> pdfs = new Vector<>();
+        for (String fileOrCiteKey : filesAndCitekeys.split(",")) {
+            if (fileOrCiteKey.toLowerCase(Locale.ROOT).endsWith(".pdf")) {
+                pdfs.add(fileOrCiteKey);
+            } else {
+                citeKeys.add(fileOrCiteKey);
+            }
+        }
+
+        writeXMPtoPdfByCitekey(databaseContext, dataBase, citeKeys, encoding, filePreferences, xmpPdfExporter);
+        writeXMPtoPdfByFileNames(databaseContext, dataBase, pdfs, encoding, filePreferences, xmpPdfExporter);
+
+    }
+
+    private void writeXMPtoPDFsOfEntry(BibDatabaseContext databaseContext, String citeKey, BibEntry entry, Charset encoding, FilePreferences filePreferences, XmpPdfExporter xmpPdfExporter) {
+        try {
+            if (xmpPdfExporter.exportToAllFilesOfEntry(databaseContext, encoding, filePreferences, entry, Arrays.asList(entry))) {
+                LOGGER.info(String.format("Successfully written XMP metadata on at least one linked file of %s", citeKey));
+            } else {
+                LOGGER.error(String.format("Cannot write XMP metadata on any linked files of %s. Make sure there is at least one linked file and the path is correct.", citeKey));
+            }
+        } catch (Exception e) {
+            LOGGER.error(String.format("Failed writing XMP metadata on a linked file of %s.", citeKey));
+        }
+    }
+
+    private void writeXMPtoPdfByCitekey(BibDatabaseContext databaseContext, BibDatabase dataBase, Vector<String> citeKeys, Charset encoding, FilePreferences filePreferences, XmpPdfExporter xmpPdfExporter) {
+        for (String citeKey : citeKeys) {
+            List<BibEntry> bibEntryList = dataBase.getEntriesByCitationKey(citeKey);
+            if (bibEntryList.isEmpty()) {
+                LOGGER.error(String.format("Skipped - Cannot find %s in library.", citeKey));
+                continue;
+            }
+            for (BibEntry entry : bibEntryList) {
+                writeXMPtoPDFsOfEntry(databaseContext, citeKey, entry, encoding, filePreferences, xmpPdfExporter);
+            }
+        }
+    }
+
+    private void writeXMPtoPdfByFileNames(BibDatabaseContext databaseContext, BibDatabase dataBase, Vector<String> fileNames, Charset encoding, FilePreferences filePreferences, XmpPdfExporter xmpPdfExporter) {
+        for (String fileName : fileNames) {
+            Path filePath = Path.of(fileName);
+            if (!filePath.isAbsolute()) {
+                filePath = FileHelper.find(fileName, databaseContext.getFileDirectories(filePreferences)).orElse(FileHelper.find(fileName, Arrays.asList(Path.of("").toAbsolutePath())).orElse(filePath));
```

Comment:
That's probably the nicer way to do it, but would that be different from Path.of("")?

Reply:
The working directory is the last directory JabRef stored something in. Path.of("") is probably the directory the JabRef executable is in.

Reply:
In that case, I don't see a benefit of using it. I think it would have been enough to allow only valid paths in the filesystem (so absolute or relative to Path.of("") ). The possibility of using the exact same path that is given in the linked file is already something that I would not expect a CLI tool to be capable of.
This is even more true for the working directory, because that would mean subsequent calls could result in different outcomes which is frowned upon.
This is kind of related to the discussion we had with @koppor on wednesday, that it is unusual to have changes in the gui (preferences) affect the behaviour of the CLI.

Reply:
Yes, that's a good point. It would provoke a major wtf moment for the user in that case.